yb-voyager/cmd/analyzeSchema.go

```diff
@@ -454,21 +454,44 @@ func checkDDL(sqlInfoArr []sqlInfo, fpath string) {
 				"", "", "FUNCTION", tbl[2], sqlInfo.formattedStmt)
 			summaryMap["FUNCTION"].invalidCount++
 		} else if regMatch := partitionColumnsRegex.FindStringSubmatch(sqlInfo.stmt); regMatch != nil {
+			// example1 - CREATE TABLE example1( 	id numeric NOT NULL, 	country_code varchar(3), 	record_type varchar(5), PRIMARY KEY (id) ) PARTITION BY RANGE (country_code, record_type) ; 
+			// example2 - CREATE TABLE example2 ( 	id numeric NOT NULL PRIMARY KEY, 	country_code varchar(3), 	record_type varchar(5) ) PARTITION BY RANGE (country_code, record_type) ; 
 			allColumns := strings.Trim(regMatch[3], "() ")
-			allColumnsList := strings.Split(allColumns, "(")
-			primaryKeyColumns := allColumnsList[len(allColumnsList)-1]
+			allColumnsList := utils.CsvStringToSlice(allColumns) 
+			// example1 - allColumnsList: [id numeric NOT NULL country_code varchar(3) record_type varchar(5) PRIMARY KEY (id]
+			// example2 - allColumnsList: [id numeric NOT NULL PRIMARY KEY country_code varchar(3) record_type varchar(5]
+			primaryKey := allColumnsList[len(allColumnsList)-1] 
+			// example1 - primaryKey: PRIMARY KEY (id
+			// example2 - primaryKey: record_type varchar(5
+			var primaryKeyColumnsList []string
+			if !strings.Contains(strings.ToLower(primaryKey), "primary key") { //true for example2 
+				//this case can come by manual intervention
+				for _, columnDefinition := range allColumnsList {
+					if strings.Contains(strings.ToLower(columnDefinition), "primary key") {
+						partsOfColumnDefinition := strings.Split(columnDefinition, " ")
+						columnName := partsOfColumnDefinition[0]
+						primaryKeyColumnsList = append(primaryKeyColumnsList, columnName) // example2 - primaryKey: [id]
```

Comment:
remove comments from every line. 
Just keep the above comments, that is good enough to understand